server/src/main/java/org/opensearch/rest/NamedRoute.java

```diff
@@ -8,18 +8,25 @@
 
 package org.opensearch.rest;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.opensearch.OpenSearchException;
+import org.opensearch.transport.TransportService;
 
 /**
  * A named Route
  *
  * @opensearch.internal
  */
 public class NamedRoute extends RestHandler.Route {
+
+    private static final Logger logger = LogManager.getLogger(NamedRoute.class);
     private static final String VALID_ACTION_NAME_PATTERN = "^[a-zA-Z0-9:/*_]*$";
     static final int MAX_LENGTH_OF_ACTION_NAME = 250;
 
-    private final String name;
+    private String name;
+
+    private String legacyName;
```

Comment:
IMO I like the name `legacyActionName` to try to convey to that this is applicable to plugins being converted to extensions. `cluster:admin/opendistro/ad/detector/info` is a TransportAction when AD is a plugin, but with AD as an extension its a RestHandler that has this legacy action name assigned to it.

Reply:
i'll update it